### PR TITLE
People: Invites: Add accepted invite welcome message

### LINKED ISSUE #133: People: Create invite accept/decline notices
At the end of the accept invite flow, we'll need to provide a UI that lets the user know that he/she successfully accepted the invite.

115107d4-82f8-11e5-9f13-44a48b7a10bb
1153134e-82f8-11e5-8678-90045e7442c1

These are notices, not a new component.

@rickybanister mentioned that it's worth searching the codebase for `Welcome`. He took that design from the welcome message shown after signing up for a new account. Something must already exist that spits it out.

### PR DESCRIPTION
**Pull request by lezama**
### How to partially test ###

* pull `add/invites-accept-decline-notices`
* go to `calypso.localhost:3000/?invite_accepted=<blog_id>`

You should see a message like this: 

image

Partially-closes #133

cc @ebinnion @roccotripaldi for an early review

### CODE REVIEW

**Comment by ebinnion** [comment]
This approach seems fine to me :+1: Although, note that there are some formatting issues currently where the notice is going behind the sidebar. This may get fixed in #302 though, so let's keep an eye on that PR.

I think we're going to run into issues with logged out users since these users won't be able to see the Calypso dashboard. So, how do we handle users that turn decline invites or users that opt to only follow by email? cc @rickybanister

**Comment by rickybanister** [comment]
If they opt to follow by email we send them to subscribe.wordpress.com.

If they decline they see something like this:

follower-flow-11

The button would take them to the normal NUX

File: client/accept-invite/actions.js [hunk 1]
```diff
@@ -1,7 +1,14 @@
 /**
```
Comment (mtias) [inline client/accept-invite/actions.js]: We should place all of these non-view pieces be in `lib`.

File: client/accept-invite/invite-message/index.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (mtias) [inline client/accept-invite/invite-message/index.jsx]: I'd consider putting the `invite-message` component in the `client/signup` folder.

File: client/layout/index.jsx [hunk 3]
```diff
@@ -107,6 +128,7 @@ module.exports = React.createClass( {
 					<Welcome isVisible={ showWelcome } closeAction={ this.closeWelcome } additionalClassName="NuxWelcome">
 						<WelcomeMessage welcomeSite={ newestSite } />
 					</Welcome>
+					{ this.getAcceptInviteMessage() }
```
Comment (mtias) [inline client/layout/index.jsx]: I think it'd be nicer if all this was absorbed in a component/sub-component to not pollute `layout/index.jsx`.

File: assets/stylesheets/_components.scss [hunk 4]
```diff
@@ -6,6 +6,7 @@
 // stylesheet order does not matter.
 // ==========================================================================
 
+@import 'accept-invite/invite-message/style';
```
Comment (mtias) [inline assets/stylesheets/_components.scss]: This should be after `invite-header` :)

**Comment by mtias** [comment]
image

Given the current message I think it should be just a success `<Notice>`.

**Comment by lezama** [comment]
@mtias could you give it another look, I changed everything :p

image

I would love some help on how to tackle the design

> I'd consider putting the invite-message component in the client/signup folder.

I think we should move the entire `accept-invite` folder on another PR

**Comment by ebinnion** [comment]
I'm not sure I agree to moving the entire directory to within `signup`. While we are using the shared signup form component, we are not at all using the same logic that the signup steps allow.

**Comment by mtias** [comment]
@ebinnion consider the large folder groups in client (my-sites, reader, me, signup, post-editor) as a way to organize the main areas of the apps and their views. There are some miscellaneous pieces (translator, layout components, etc) that don't quite fit in any of these buckets. Perhaps this is one of them? I'd like for us to find a place for the react components used here that is not the root of client. We've had on our minds the idea of reworking the `layout` folder into a more general `app` one once we move into a single render tree.

Now thinking a bit more about this, `invites` seems like a "my-sites" domain in general. Why not move it there?

File: client/accept-invite/invite-message/index.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		}
+		const { accepted, declined, siteId } = this.state;
+		if ( accepted ) {
+			const site = this.props.sites.getSite( siteId );
+			if ( ! site ) {
+				return null;
+			}
+			return (
+				<SimpleNotice status="is-success" onClick={ dismissInviteAccepted }>
+					<h3 className="invite-message__title">{ this.translate( 'You\'re now a user of: %(site)s', { args: { site: site.slug } } ) }</h3>
```
Comment (mtias) [inline client/accept-invite/invite-message/index.jsx]: I think the site should be a link to the front-end. Also the sentence would read better without the `:` — `You're now a user of <a>Blabla</a>.`, with a period at the end.

**Comment by mtias** [comment]
> I think we should move the entire accept-invite folder on another PR

Sure, but please, open an issue for it so we don't lose track — I hope we avoid adding stuff into client's root in general.

**Comment by ebinnion** [comment]
cc @rickybanister for a design review.

File: client/accept-invite/actions.js [hunk 6]
```diff
@@ -2,6 +2,8 @@
  * Internal dependencies
  */
 import wpcom from 'lib/wp' ;
+import Dispatcher from 'dispatcher';
+import { DISPLAY_INVITE_ACCEPTED_NOTICE, DISMISS_INVITE_ACCEPTED_NOTICE, DISPLAY_INVITE_DECLINED_NOTICE, DISMISS_INVITE_DECLINED_NOTICE } from './invite-message/constants'
```
Comment (johnHackworth) [inline client/accept-invite/actions.js]: Oooh, I love this. Great idea!! I'm going to copy it :D

File: client/accept-invite/invite-message/index.jsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import React from 'react'
+
+/**
+ * Internal dependencies
+ */
+import SimpleNotice from 'notices/simple-notice'
```
Comment (johnHackworth) [inline client/accept-invite/invite-message/index.jsx]: I think SimpleNotice have just been (like... 2 hours ago) deprecated! :/

**Comment by johnHackworth** [comment]
We need to update this for using <Notice> instead of <SimpleNotice>, but once that is done, I think code-wise this is ready to go :shipit:

**Pull request merged**